### ISSUE TITLE
view.set around onChange in deep sync

### ISSUE THREAD

**Issue by ncammarata** [flintjs/flint#192]
variables work fine
```js
view Main {
  let name = "nick"
  <h1>Hello {name}</h1>
  <button onClick={() => name='hi'}>change</button>
  <input sync={name} />
}
```

creates the attribute

```js
"__flintOnChange": function (e) {view.set("name", name = e.target.value)} 
```

deep variables 
```js
view Main {
  let name = "nick"
  <h1>Hello {name}</h1>
  <button onClick={() => name='hi'}>change</button>
  <input sync={name} />
}
```

don't wrap with view.set

```js
```"__flintOnChange": function (e) {person.name = e.target.value} ```
```

thanks to @ZZShark9 for the find

**Comment by natew** [comment]
I'm getting (input):

```js
view Main {
  <input sync={person.name} />
}
```

Output:

```js
Flint.view("Main", function (view, on, $) {
  view.render(function () {
    return React.createElement(["input", 1], {
      "__flintValue": person.name,
      "__flintOnChange": function (e) {
        person.name = e.target.value
      }
    });
  });
});
```

**Comment by ncammarata** [comment]
closed in a006f2693b0a19e5cd852d213af2d17e3b25e26e
